### PR TITLE
Text claiming all pinned variants are represented in causatives is now only rendered if this is true

### LINKED ISSUE #4547: case_report renders wrong text for contents
When the card for causative variants has no contents but the card for pinned variants do, the pinned variants card is ended with a text stating "All pinned variants are described among the causative variants"

image

However, when the pinned variants are indeed present among the causative variants the card is rendered empty:

image

### PR DESCRIPTION
**Pull request by TereseBo**
This PR fixes a bug which caused the text "All pinned variants are described among the causative variants" to be displayed when this was not the case.

This PR closes #4547

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by TB

### CODE REVIEW

**Comment by TereseBo** [comment]
On main branch, when the pinned variants are not found in the causatives:

image

cancer case
text claiminng all pinned are in causatives

constitutional case, the card is rendered empty when all variants are in causatives:
image


On this branch, with the same variants in causatives and pinned:

cancer case
image

constitutional case:
image

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 1]
```diff
@@ -342,13 +342,15 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
   </div>
   <div class="card-body">
     {% set duplicated_variants = [] %}
+    {% set pinned_not_causative_variants = [] %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html:345]: 👍

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 2]
```diff
@@ -358,7 +360,7 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
     {% else %}
       No pinned variants available for this case
     {% endif %}
-    {% if variants.suspects_detailed and duplicated_variants|length == variants.suspects_detailed.suspects_detailed|length %}
+    {% if pinned_not_causative_variants|length == 0 %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html:363]: Yep, more direct and actually correct. 😄

**Review by dnil — APPROVED** [review]
🎉

File: scout/server/blueprints/cases/templates/cases/case_report.html (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/case_report.html:363]: Ah, hm, well, actually checking for any pinned would be pretty good as well, or you would end up with No pinned vars - All pinned vars are causative..

**Pull request merged**